src/tools/pkcs15-tool.c

```diff
@@ -982,6 +982,67 @@ static int read_ssh_key(void)
 		return 1;
 	}
 
+	if (pubkey->algorithm == SC_ALGORITHM_EC) {
+		// support only for NIST
+		// 'ssh-keygen -t ecdsa' allow only field lengths 256/384/521
+
+		static struct supported_ec_curves {
+			char *curve_name;
+			struct sc_object_id curve_oid;
+			size_t size;
+		} ec_curves[] = {
+			{"secp256r1", {{1, 2, 840, 10045, 3, 1, 7, -1}},256},
+			{"secp384r1", {{1, 3, 132, 0, 34, -1}},         384},
+			{"secp521r1", {{1, 3, 132, 0, 35, -1}},         521},
+		        {NULL, {{-1}}, 0},
+		};
+		char alg[20];
+		unsigned char buf[4+30+4+9+4+256];
```

Comment:
I would write there either "big enough" number as for the RSA section or add there some comment why specially these numbers. Otherwise the code becomes it is very hard to maintain.

Reply:
Ok, what about:

#define ITEM_SIZE 4
#define ALG_NAME 30
#define CURVE_NAME 9
#define  KEY_DATA 256
..
unsigned char buf[3*ITEM_SIZE+ALG_NAME+CURVE_NAME+KEY_DATA];

???

Reply:
Well ... you already propose that 30 chars is enough to fit `ecdsa-sha2-nistp521` (19 chars), so I would rather put there something like this:
```
/* Large enough to fit the following:
 * 3 x 4B item length headers
 * max 20B algorithm name,  9B curve name, max 256B key data */
unsigned char buf[300];
```
Using constants makes it clearer, but significantly longer.

Reply:
Thanks for hint, pull request is already updated as your suggested.